### ISSUE TITLE
Include stats about repository views and size

### ISSUE THREAD

**Issue by mike-r-mclaughlin** [sourcegraph/sourcegraph#40317]
#### Feature request description

Include usage stats for each repository along with the repository size with a view sortable by size and usage.  Administrators can use this view to help them determine if it is worth it to keep a very large (and likely resource costly) repository stored in Sourcegraph.

#### Is your feature request related to a problem? If so, please describe.

There is always a battle for resources (CPU/memory/disk) especially as more groups want to onboard repositories.  It would be great if we could find the repositories that are taking up a lot of resources and remove/exclude those so there's room for other teams' repos.

#### Describe alternatives you've considered.

Write a script

#### Additional context

Requested by: https://github.com/sourcegraph/accounts/issues/6716

**Comment by vdavid** [comment]
I'm picking this up partially: I'm adding sorting by **size**. After some discussion with @mrnugget, we don't seem to have a good definition for "usage", and "repository views" (if we mean that by "usage") is something we don't track at the moment. So it makes sense to me to break this up into two deliverables and deliver the smaller part.

I have two possible solutions in mind:
1. 🪨 **Purist**: Add a checkbox to set “Sort by size, decreasing”, release, done.
2. ✨ **Elegant**: Reuse @erzhtor and @thenamankumar's “UsersList” / “Table” components from the [Users](https://sourcegraph.sourcegraph.com/site-admin/users) page. This has the nice side-effect of making this component more available for reuse at other places.

I prefer no. 2. Yesterday I got good news from Erzhan and Naman: “The table component we created is fairly generic and can be reused easily. You will need to extract it out of the User Management directory, which isn't a big deal. [...] Configuring sorting is pretty easy as well.”, plus they offered help.

As a next step, I'll look into their code, and if needed, set up a meeting with Naman to plan reusing it.

Note: If going with solution no. 2., the additional benefit would be sorting by name, last sync date, and shard index, plus easier extensibility with filtering.

For reference, quick links to the [Repo admin](https://sourcegraph.sourcegraph.com/site-admin/repositories) and the [User admin](https://sourcegraph.sourcegraph.com/site-admin/users) pages on S2
